src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -325,6 +332,25 @@ SyncMetadataSnapshotToNode(WorkerNode *workerNode, bool raiseOnError)
 }
 
 
+/*
+ * DropMetadataSnapshotOnNode creates the queries which drop the metadata and sends them
+ * to the worker given as parameter.
+ */
+static void
+DropMetadataSnapshotOnNode(WorkerNode *workerNode)
+{
+	char *extensionOwner = CitusExtensionOwnerName();
+
+	/* generate the queries which drop the metadata */
+	List *dropMetadataCommandList = MetadataDropCommands();
+
+	SendOptionalCommandListToWorkerInTransaction(workerNode->workerName,
```

Comment:
Is there a specific reason why we use `SendOptionalCommandListToWorkerInTransaction()` API?

With this one, if an operation fails on the remote node, we do not rollback the transaction  on the coordinator. 

I think we should be using `SendCommandListToWorkerInSingleTransaction()` which fails the transaction on the coordinator as well. This seems more desirable user experience.